power-control-x86/src/power_control.cpp

```diff
@@ -614,50 +614,48 @@ static void powerRestorePolicyCheck()
                 const std::variant<std::string>& restorepolicy) {
         if (ec)
         {
+            std::cerr << "ec Error in power restore policy check\n";
             return;
         }
         const std::string* policy =
                 std::get_if<std::string>(&restorepolicy);
         if (policy == nullptr)
         {
-            std::cerr << "Unable to read restore policy\n";
+            std::cerr << "Unable to read power restore policy\n";
             return;
         }
 
-        if (policy->compare("None") == 0)
+        std::cerr << "power restore policy is " << policy->c_str() << " \n";
+
+        if (policy->compare("xyz.openbmc_project.Control.Power.RestorePolicy.Policy.AlwaysOn") == 0)
         {
-            return;
+            std::cerr << "Restore policy to Alwaya On\n";
+            sendPowerControlEvent(Event::powerOnRequest);
+            setRestartCauseProperty(getRestartCause(RestartCause::powerPolicyOn));
         }
-        else
+        else if (policy->compare("xyz.openbmc_project.Control.Power.RestorePolicy.Policy.Restore") == 0)
         {
-            std::cerr << "Invoking Restore Policy: " << policy << "\n";
-
-            sd_journal_send("MESSAGE=PowerControl: power restore policy applied",
-                    "PRIORITY=%i", LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-                    "OpenBMC.0.1.PowerRestorePolicyApplied", NULL);
-
-            if (policy->compare("AlwaysOn") == 0)
```

Comment:
Change this condition.

Reply:
it is done with:
if (policy->compare("xyz.openbmc_project.Control.Power.RestorePolicy.Policy.AlwaysOn") == 0)